### PR TITLE
[generated] source: spec3.sdk.yaml@spec-6c38dc0 in master

### PR DESCRIPTION
**Pull request by mickjermsurawong-stripe**
This PR adds support for:
- extra params at all level of params object.
- 3 additional features:
  1) webhook event payment_method.updated
  2) charge list filter to contain payment intent
  3) account additional enums legacy_payments

- Tests for the basic primitives for handling extra params are in the base PR https://github.com/stripe/stripe-java/pull/773

r? @remi-stripe for the new additional features please
r? @ob-stripe for review over the extra params please
cc @stripe/api-libraries

### CODE REVIEW

File: src/main/java/com/stripe/param/WebhookEndpointCreateParams.java [hunk 1]
```diff
@@ -683,6 +729,9 @@ public Builder setUrl(String url) {
     @SerializedName("payment_method.detached")
     PAYMENT_METHOD__DETACHED("payment_method.detached"),
 
+    @SerializedName("payment_method.updated")
+    PAYMENT_METHOD__UPDATED("payment_method.updated"),
```
Comment (mickjermsurawong-stripe) [inline src/main/java/com/stripe/param/WebhookEndpointCreateParams.java]: 1) Webhook event enum for payment method

File: src/main/java/com/stripe/param/ChargeListParams.java [hunk 2]
```diff
@@ -28,13 +30,28 @@
...
    * default is 10.
    */
   @SerializedName("limit")
   Long limit;
 
+  /**
+   * Only return charges that were created by the PaymentIntent specified by this PaymentIntent ID.
+   */
+  @SerializedName("payment_intent")
+  String paymentIntent;
```
Comment (mickjermsurawong-stripe) [inline src/main/java/com/stripe/param/ChargeListParams.java]: 2) Charge list filter with payment intent

File: src/main/java/com/stripe/param/AccountCreateParams.java [hunk 3]
```diff
@@ -2547,6 +3430,9 @@ public Builder setUserAgent(String userAgent) {
     @SerializedName("card_payments")
     CARD_PAYMENTS("card_payments"),
 
+    @SerializedName("legacy_payments")
+    LEGACY_PAYMENTS("legacy_payments"),
```
Comment (mickjermsurawong-stripe) [inline src/main/java/com/stripe/param/AccountCreateParams.java]: 3) Account requested capability with `legacy_payments`

**Comment by remi-stripe** [comment]
All good from me on the additional features since I approved them separately!

**Comment by mickjermsurawong-stripe** [comment]
thank you @ob-stripe and @remi-stripe!
Merging this to the integration branch

**Pull request merged**